### ISSUE TITLE
Use seed nodes as trusted peeers while initializing go-header

### ISSUE THREAD

**Issue by tzdybal** [rollkit/rollkit#1714]
TL;DR: we should use seed nodes as trusted peers while initializing go-header. This is in-line with usage in celestia-node, where bootstrappers are used as trusted peers.

----

While working on https://github.com/rollkit/rollkit/pull/1684 I discovered some issues related to peer handling in `go-header`. Those issues are on rollkit side - `go-header` is not used as it supposed to be.

They are visible only on latest releases of `go-libp2p` and `go-header` because of some slight changes related to peer discovery, timing, etc.

Currently full nodes try to discover peers using DHT, and then use them as trusted peers while initializing go-header.
If peers are not discovered properly (and it seems a bit more time is needed with latest libp2p versions) before go-header initialization, `go-header` `Syncer` is not started at all.

To fix this, we should feed our seed nodes as trusted peers for `go-header`.
